tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/UpdateCombinationStockFeatureContext.php

```diff
@@ -0,0 +1,141 @@
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class UpdateCombinationStockFeatureContext extends AbstractCombinationFeatureContext
+{
+    /**
+     * @When I update combination :combinationReference stock with following details:
+     *
+     * @param string $combinationReference
+     * @param TableNode $tableNode
+     */
+    public function updateStock(string $combinationReference, TableNode $tableNode): void
+    {
+        $command = new UpdateCombinationStockCommand($this->getSharedStorage()->get($combinationReference));
+        $this->fillCommand($command, $tableNode->getRowsHash());
+
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @Transform table:combination stock detail,value
+     *
+     * @param TableNode $tableNode
+     *
+     * @return CombinationStock
+     */
+    public function transformCombinationStock(TableNode $tableNode): CombinationStock
+    {
+        $dataRows = $tableNode->getRowsHash();
+
+        return new CombinationStock(
+            (int) $dataRows['quantity'],
+            (int) $dataRows['minimal quantity'],
+            $dataRows['location'],
+            (int) $dataRows['low stock threshold'],
+            PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['low stock alert is on']),
+            PrimitiveUtils::castElementInType($dataRows['available date'], PrimitiveUtils::TYPE_DATETIME)
+        );
+    }
+
+    /**
+     * @Then combination :combinationReference should have following stock details:
+     *
+     * @param string $combinationReference
+     * @param CombinationStock $expectedStock
+     *
+     * @throws CombinationConstraintException
+     */
+    public function assertStockDetails(string $combinationReference, CombinationStock $expectedStock): void
+    {
+        $actualStock = $this->getCombinationForEditing($combinationReference)->getStock();
+
+        Assert::assertSame(
+            $expectedStock->getQuantity(),
+            $actualStock->getQuantity(),
+            sprintf('Unexpected combination "%s" quantity', $combinationReference)
+        );
+        Assert::assertSame(
+            $expectedStock->getMinimalQuantity(),
+            $actualStock->getMinimalQuantity(),
+            sprintf('Unexpected combination "%s" minimal quantity', $combinationReference)
+        );
+        Assert::assertSame(
+            $expectedStock->getLowStockThreshold(),
+            $actualStock->getLowStockThreshold(),
+            sprintf('Unexpected combination "%s" low stock threshold', $combinationReference)
+        );
+        Assert::assertSame(
+            $expectedStock->getLocation(),
+            $actualStock->getLocation(),
+            sprintf('Unexpected combination "%s" location', $combinationReference)
+        );
+        Assert::assertEquals(
+            $expectedStock->getAvailableDate()->format(DateTimeUtil::DEFAULT_DATETIME_FORMAT),
```

Comment:
I just discovered a behaviour with specific price When a date is saved as 0 so it matches `DateTime::NULL_VALUE = '0000-00-00 00:00:00'` then the previous controller used `null` as value Which allows to leave the form input empty

I think it's a good idea and it probably should be applied in this case as well, `getAvailableDate()` should be nullable The same thing is appliable in `ProductStockInformation` as well but I didn't know at the time

Reply:
That's how I handled it for specific prices https://github.com/PrestaShop/PrestaShop/pull/22280
For the test data I considered empty value to be expected as `null` for DateTime fields

Reply:
yes, this requires to use string as command input, if that's acceptable then im ok with it :man_shrugging: 
But thn I suppose we should use strings instead of DateTime as command inputs everywhere

Reply:
Actually it's more on the query side that we need to handle this, because the DB doesn't store null it stores `0000-00-00 00:00:00` So the main modification is in the Query to translate `0000-00-00 00:00:00` value from the DB as `null` (this means we can't save the `0000-00-00 00:00:00` value but we are limited by our database structure here because fields are not nullable and automatically filled by ObjectModel with `0000-00-00`)

Regarding the command input I do wonder if we should use DateTime as input or not, it's quite similar to the use of DecimalNumber or not? 🤷‍♂️  I would tend to use DateTime because it's more accurate since you can also provide a timezone which will have impact on the true DateTime value (especially its timestamp), but it doesn't seem to be something really managed in the core code for now

Reply:
sorry i don't get it.
Problem with queries already seems to be solved by using that NULL_VALUE (0000-00-00) from Utils as you mentioned.

The problem was indeed the input of command, because if we use DateTime objects we cannot pass string (0000-00-00 00:00:00) which supposed to be considered null (we cannot pass null itself either as its already used for identifying if property is updated).

Reply:
No but we can use `null` instead when the DB value is `0000-00-00 00:00:00` This needs to be handled in the query handler. See this example from my PR:
https://github.com/PrestaShop/PrestaShop/blob/acc0e2d1857a060a7e1c21a33ce9e3a1e7d68c4e/src/Adapter/Product/QueryHandler/GetProductForEditingHandler.php#L331

This was inspired by what was already done in:

https://github.com/PrestaShop/PrestaShop/blob/3f84ef894cd39a9284ab399d103b54b290b6a373/src/PrestaShopBundle/Controller/Admin/SpecificPriceController.php#L271